proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,243 @@
+# Test Plan
+
+All tests are to be included in the HLK test binary which ships with the OS.
+This test binary is only built in the OS repo and based off of the 'ExecTests'
+'ExecTests' source code in the DXC repo. There is a script in the WinTools repo
+which generates and annotates the HLK tests.
+
+There are three test categories we are concerned with:
```

Comment:
Do we need special tests for groupshared?

Are there any other types of storage that might be interesting?

Reply:
Probably, I was thinking about this and I need to learn/explore more. I suspect I'm actually missing details on test case granularity. Does testing across data types matter? Do we care to verify the output of operations for correctness?

Reply:
We should probably be testing for correctness, yes.  Sizes, especially around interesting alignment cases, are probably interesting.  It might be interesting to build cases that could conceivably generate overruns and check for them.  For example, most GPUs will operate on at least 32-bits at once, so if you have 16-bit values then what happens if you have an odd number of elements - could it accidentally overwrite the next variable if it is assuming alignment?

Reply:
And so the size of a type, and the element count, gets interesting.  I expect you could come up with some interesting boundary cases and reuse that across all the types without having to think hard about each combination individually.